tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/OrderExtensionsTests.cs

```diff
@@ -130,5 +130,99 @@ public void IsAssociatedServicesSectionComplete_ToggleViewed_ReturnsBool(bool vi
             var actual = order.IsAssociatedServicesSectionComplete();
             actual.Should().BeFalse();
         }
+
+        [Test]
+        public void IsFundingSourceComplete_OrderFundingViewed_ReturnsTrue()
+        {
+            var order = OrderBuilder.Create().WithFundingSourceOnlyGms(true).Build();
+            var actual = OrderExtensions.IsFundingSourceComplete(order);
+            actual.Should().BeTrue();
+        }
+
+        [Test]
+        public void IsFundingSourceComplete_NullOrder_ReturnsFalse()
+        {
+            var actual = OrderExtensions.IsFundingSourceComplete(null);
+            actual.Should().BeFalse();
+        }
+
+        [Test]
+        public void IsSectionStatusCompleteComplete_NullOrder_ReturnsFalse()
+        {
+            var actual = OrderExtensions.IsSectionStatusComplete(null,0,0,0);
+            actual.Should().BeFalse();
+        }
+
+        [Test]
+        public void IsSectionStatusCompleteComplete_IncompleteOrder_ReturnsFalse()
+        {
+            var order = OrderBuilder
+                .Create()
+                .WithFundingSourceOnlyGms(null)
+                .WithServiceRecipientsViewed(false)
+                .WithAssociatedServicesViewed(false)
+                .WithCatalogueSolutionsViewed(false)
+                .Build();
+
+            var actual = OrderExtensions.IsSectionStatusComplete(order, 0, 0, 0);
+            actual.Should().BeFalse();
+        }
+
+
+        [Test]
+        public void IsSectionStatusCompleteComplete_FundingCompleteCatalogSolutionCompleteOrder_ReturnsTrue()
```

Comment:
Would it be possible to combine some of these tests by using the TestCase attribute? There might be a lot of parameters that might make the test a little unmanageable but see what you think.